webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/campaigns/MainCampaignService.scala

```diff
@@ -75,6 +76,17 @@ class MainCampaignService(repo: CampaignEventRepository, campaignRepo: CampaignR
 
   private[this] var inner: Option[CampaignScheduler] = None
 
+  def deleteCampaign(c: CampaignId) = {
+
+    inner match {
+      case Some(s) => for {
+        _ <- s.deleteCampaign(c)
+        _ <- campaignRepo.delete(c)
+      } yield {  () }
+      case None    => Inconsistency("not initialized yet").fail
```

Comment:
I think in that case we want to be idempotent and just say that everything is OK (perhaps in log debug say that the event was not found)